arches/app/media/js/viewmodels/reference-select.js

```diff
@@ -0,0 +1,160 @@
+define([
+    'jquery',
+    'knockout',
+    'knockout-mapping',
+    'arches',
+    'viewmodels/widget',
+], function($, ko, koMapping, arches, WidgetViewModel) {
+    var NAME_LOOKUP = {};
+    var ReferenceSelectViewModel = function(params) {
+        var self = this;
+
+        params.configKeys = ['placeholder'];
+        this.multiple = !!ko.unwrap(params.node.config.multiValue);
+        this.displayName = ko.observable('');
+        this.selectionValue = ko.observable([]); // formatted version of this.value that select2 can use
+        this.activeLanguage = arches.activeLanguage;
+
+        WidgetViewModel.apply(this, [params]);
+
+        this.getPrefLabel = function(labels){
+            return koMapping.toJS(labels)?.find(label => label.language===arches.activeLanguage && label.valuetype === 'prefLabel')?.value || arches.translations.unlabeledItem;
+        }; 
+
+        this.displayValue = ko.computed(function() {
+            const val = self.value();
+            let name = '';
+            if (val) {
+                name = val.map(item=>self.getPrefLabel(item.labels)).join(", ");
+            }
+            return val ? name : null;
+        });
+
+        this.valueAndSelectionDiffer = function(value, selection) {
+            if (!(ko.unwrap(value) instanceof Array)) {
+                return true;
+            }
+            const valueUris = ko.unwrap(value).map(val=>ko.unwrap(val.uri));
+            return (JSON.stringify(selection) !== JSON.stringify(valueUris))
```

Comment:
> a couple minor changes to ensure the dirty state clears properly if a user choses the same options with which the tile loaded

Thanks, I'll be sure to update that again after #10602, since that adds more fields.

Your point reminds me that .stringify isn't the best choice here, as JSON object keys are unordered. Python dicts are ordered (now), but nothing on the JS side is. I can open a follow-up issue if you prefer.

Reply:
I misstated things, looks like `JSON.stringify()` has a deterministic ordering as of ES2015, not sure about response.json() though. Maybe we should just sort right before this comparison.

Reply:
Yeah - a quick sort would be the safe thing to do.

Reply:
Based on the experience with the resource instance widget this is pretty hypothetical